app/assets/javascripts/covid19/src/webapp/components/structures-table/CustomCheckboxFilter.tsx

```diff
@@ -1,54 +1,95 @@
+import styled from "styled-components";
 
 export interface CustomCheckboxFilterProps {
-    filterState: EntityBodiesFilter;
-    setFilterState(filter: EntityBodiesFilter): void;
+    filterState: Covid19Filter;
+    setFilterState(filter: Covid19Filter): void;
 }
 
 export const CustomCheckboxFilter: React.FC<CustomCheckboxFilterProps> = React.memo(props => {
     const { filterState, setFilterState } = props;
+    const [anchorEl, setAnchorEl] = React.useState(null);
+    const isOpen = Boolean(anchorEl);
 
+    const openMenu = React.useCallback(event => setAnchorEl(event.currentTarget), []);
+    const closeMenu = React.useCallback(() => setAnchorEl(null), []);
     const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
         setFilterState({
             ...filterState,
             [event.target.name]: event.target.checked,
         });
     };
+    const startIcon = React.useMemo(() => <FilterListIcon />, []);
 
     return (
-        <FormGroup style={{ display: "flex", flexDirection: "row", marginLeft: 15 }}>
-            <FormControlLabel
-                control={
-                    <Checkbox
-                        checked={filterState?.antibody || false}
-                        onChange={handleChange}
-                        name="antibody"
-                    />
-                }
-                label="Antibodies"
-            />
-            <FormControlLabel
-                control={
-                    <Checkbox
-                        checked={filterState?.nanobody || false}
-                        onChange={handleChange}
-                        name="nanobody"
-                    />
-                }
-                label="Nanobodies"
-            />
-            <FormControlLabel
-                control={
-                    <Checkbox
-                        checked={filterState?.sybody || false}
-                        onChange={handleChange}
-                        name="sybody"
-                    />
-                }
-                label="Sybodies"
-            />
-        </FormGroup>
+        <React.Fragment>
+            <StyledSearchExampleButton
+                color="primary"
+                size="small"
+                onClick={openMenu}
+                aria-expanded={Boolean(anchorEl)}
+                aria-label="toolbarExportLabel"
+                aria-haspopup="menu"
+                endIcon={startIcon}
+                style={{ margin: "auto 5px" }}
+            >
+                {i18n.t("Filter")}
+            </StyledSearchExampleButton>
+
+            <GridMenu
+                open={isOpen}
+                target={anchorEl}
+                onClickAway={closeMenu}
+                position="bottom-start"
+            >
+                <MenuList className="MuiDataGrid-gridMenuList" autoFocusItem={isOpen}>
+                    <MenuItem>
+                        <StyledCheckbox
+                            checked={filterState?.antibodies || false}
```

Comment:
Maybe at some point `filterState` could be `undefined`, but accordingly to TS, not anymore. So: `filterState.antibodies` should work, right?